lib/fog/vsphere/requests/compute/list_vm_cdroms.rb

```diff
@@ -0,0 +1,26 @@
+module Fog
+  module Compute
+    class Vsphere
+      class Real
+        def list_vm_cdroms(vm_id)
+          get_vm_ref(vm_id).config.hardware.device.select { |hw| hw.class == RbVmomi::VIM::VirtualCdrom }.map do |cdrom|
+            {
+              :filename => (cdrom.backing.fileName rescue(nil)),
+              :name => cdrom.deviceInfo.label,
+              :key => cdrom.key,
+              :controller_key => cdrom.controllerKey,
+              :unit_number => cdrom.unitNumber,
+              :start_connected => cdrom.connectable.startConnected,
+              :allow_guest_control => cdrom.connectable.allowGuestControl,
+              :connected => cdrom.connectable.connected,
+            }
+          end
+        end
+      end
+      class Mock
+        def list_vm_cdroms(vm_id)
+        end
+      end
```

Comment:
If you are not going to implement it don't define it because fog will raise an exception out of the box saying it is not defined which is far better then doing nothing.